src/library/Utilities/RemoteContainerConnectionDetailsResolver.cs

```diff
@@ -49,6 +49,17 @@ internal class RemoteContainerConnectionDetailsResolver
                 return remoteContainerConnectionDetails;
             }
 
+            // Check if restoration pod is present. This is an indication that previous session is still connected or it has not finished restoring yet.
+            var allPods = (await _kubernetesClient.ListPodsInNamespaceAsync(remoteContainerConnectionDetails.NamespaceName, null, cancellationToken))?.Items;
+            var podsRunningRestorationJob = allPods.Where(p => p.Metadata.Name.StartsWith(remoteContainerConnectionDetails.ServiceName + "-restore") &&

```

Comment:
💡 Thoughts/A quick note: there are few places where we are using `?` for nullable type check, in case if they are nullable will it be worth logging it? example: `p.Status?.ContainerStatuses?`  (I am not sure of use case hence just idea)

if it is worth to log might be use `HasValue` and log that those value are null etc.

Reply:
I think it is safety checks because of how many different service / deployment configurations different users can have. I think if we log every time we have one of those checks the logs would be too noisy and the code more complex (since we would have several more if else levels).

Reply:
It might also be a good idea to check that the pod status is running. If the pod has crashed, starting a new debug session will redeploy the restore pod and fix the problem today.

Reply:
@Tatsinnit, sorry I misunderstood you. I see what you are saying that if pod is there, then status should never be null. I took usage from existing code previous team had writing, and without knowing historical reason on why they added it to begin with, I will leave it for now.

@Eneuman, this is good point. I am adding that check now.